### ISSUE TITLE
undefined method `map' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by neidiom** [seanhuber/rsync_wrapper#2]
Hi @seanhuber 

I am using your default example and i am getting following error.

```
/Users/nedim/.rvm/gems/ruby-2.4.1/gems/rsync_wrapper-1.0.0/lib/rsync_wrapper/rsync.rb:8:in `initialize': undefined method `map' for nil:NilClass (NoMethodError)
	from r.rb:6:in `new'
	from r.rb:6:in `<main>'
```

**Comment by seanhuber** [comment]
Thanks for reporting; this does look like a valid bug.  I'll get this patched soon.  Would you mind trying this variant of the example and see if it works as expected:

```ruby
source_directory = '/some/source/directory'
destination_directory = '/some/destination/directory'

rsync = Rsync.new(src_dir: source_directory, dest_dir: destination_directory, include_extensions: [:*])

rsync.sync! do |file_path, new_file|
  puts "Full Path: #{File.join(destination_directory, file_path)}"
  puts "The relative path of the new file is: #{file_path}"
  if new_file
    puts "This is a brand new file."
  else
    puts "This is an updated file."
  end
end
```

**Comment by neidiom** [comment]
@seanhuber Thanks for your support. Running the code you provided I am getting the error below.

```
/Users/nedim/.rvm/gems/ruby-2.4.1/gems/rsync_wrapper-1.0.0/lib/rsync_wrapper/rsync.rb:34:in `initialize': no implicit conversion of nil into String (TypeError)
	from /Users/nedim/.rvm/gems/ruby-2.4.1/gems/rsync_wrapper-1.0.0/lib/rsync_wrapper/rsync.rb:34:in `open'
	from /Users/nedim/.rvm/gems/ruby-2.4.1/gems/rsync_wrapper-1.0.0/lib/rsync_wrapper/rsync.rb:34:in `parse_logfile'
	from /Users/nedim/.rvm/gems/ruby-2.4.1/gems/rsync_wrapper-1.0.0/lib/rsync_wrapper/rsync.rb:19:in `sync!'
	from r2.rb:8:in `<main>'
```

**Comment by seanhuber** [comment]
Okay I was able to replicate the problem and coded a fix for it.  The patch has been released in version 1.0.1 of this gem.  Please upgrade (`bundle update rsync_wrapper`) and try the original example again.  Thanks!

050b31fc4941009d10f5509d1bc4c94d3fc7c37b
